### PR TITLE
Update 1.2 changelog to follow release tools format

### PR DESCRIPTION
**Pull request by msau42**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
Used kubernetes release-note tool 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: CHANGELOG-1.2.md [hunk 1]
```diff
@@ -1,3 +1,12 @@
+# Changelog since v1.2.0
+
+## Notable Features
+
+* Adds CSI Migration support for Azure Disk/File, adds some backwards compatibility fixes for GCE PD Migration ([#292](https://github.com/kubernetes-csi/external-provisioner/pull/292), [@davidz627](https://github.com/davidz627))
```
Comment (davidz627) [inline CHANGELOG-1.2.md]: technically this PR https://github.com/kubernetes-csi/external-provisioner/pull/294 (?) I'm not clear on which CHANGELOGS go where and whether we refer to master branch or cherry picks to release branches

Comment (msau42) [reply CHANGELOG-1.2.md]: Hm I guess it should use cherry-pick PRs instead of master ones.

Comment (msau42) [reply CHANGELOG-1.2.md]: Fixed

**Pull request merged**